### PR TITLE
Add javadoc comments to openj9 modules

### LINKED ISSUE #6951: OpenJ9 modules have no javadoc
javadoc treats the comment immediately preceding the module definition in module-info.java as the doc for the module - see https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html. For the openj9.xxxx modules (most of them anyway), the only comment ahead of the module definition is the EPLv2 license, which therefore becomes the module description.

### PR DESCRIPTION
**Pull request by pshipton**
Fixes #6951

Signed-off-by: Peter Shipton <Peter_Shipton@ca.ibm.com>

### CODE REVIEW

File: jcl/src/openj9.gpu/share/classes/module-info.java [hunk 1]
```diff
@@ -23,6 +23,10 @@
 
 /*[REM] This file must not use tabs because the dependency recognition code in openjdk does not support them. */
 
+/**
+ * Defines an API to perform certain operations using any connected CUDA capable GPU,
```
Comment (keithc-ca) [inline jcl/src/openj9.gpu/share/classes/module-info.java]: I think we should remove 'an' for consistency with other modules.

Comment (pshipton) [reply jcl/src/openj9.gpu/share/classes/module-info.java]: fixed

**Comment by pshipton** [comment]
Note that once this is approved I'll create a PR for the 0.17 release as well.

**Comment by keithc-ca** [comment]
Jenkins compile plinux jdk11

**Comment by pshipton** [comment]
Tentatively created https://github.com/eclipse/openj9/pull/7317, a cherry pick of this commit against the 0.17 branch.

**Pull request merged**